[CMS] Validation messages missing from datepicker form elements

vasilii-b opened the pull request:
### Description (*)
This PR adds both frontend and backend validation for the "Custom Theme" From-To date range.
The original PR #17520 was created to be merged to `2.2-develop` and is inactive for a while. Moreover, there is not added the backend validation.

![xnip2019-01-30_11-26-45](https://user-images.githubusercontent.com/13456702/51972642-8fe5dd00-2484-11e9-9cb6-2f3ee75dc7b9.jpg)

![2019-01-30 11 34 30](https://user-images.githubusercontent.com/13456702/51972656-9bd19f00-2484-11e9-9744-9a2cac9471fb.gif)


### Fixed Issues (if relevant)
1. magento/magento2#3795 - Validation messages missing from datepicker form elements

### Manual testing scenarios (*)
Provided in task description

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
magento/magento2#3795 - Validation messages missing from datepicker form elements

Added from-to date validation to CMS Page

magento-engcom-team commented:
Hi @vasilii-b. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

Commit:
CMS Page custom theme from-to date validation

Made newly added validation method private instead of protected

rogyar commented on app/code/Magento/Cms/Model/ResourceModel/Page.php:
```diff
@@ -265,6 +273,28 @@ protected function isValidPageIdentifier(AbstractModel $object)
+     * @param AbstractModel $object
+     * @return bool
+     */
+    private function isValidPageCustomThemeDateRange(AbstractModel $object)
```
One more thing, should we clarify the type hinting here by using page model instead fo AbstractModel?

vasilii-b replied:
Hi @rogyar, I've changed it to the CMS Page. Thank you!

Commit:
magento/magento2#3795 - Validation messages missing from datepicker form elements

- Fixed small typo
- Changed type hint for `isValidPageCustomThemeDateRange` method

rogyar commented:
@vasilii-b could you check the failing tests, please? Thank you

sidolov commented:
@vasilii-b , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-validation-message-missing-from-datepicker-form-elements

# Conflicts:

#app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js

VladimirZaets commented:
@vasilii-b, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration
